scene/resources/resource_format_text.cpp

```diff
@@ -1398,6 +1398,62 @@ ResourceUID::ID ResourceLoaderText::get_uid(Ref<FileAccess> p_f) {
 	return ResourceUID::INVALID_ID;
 }
 
+// Please call ResourceLoaderText::open() before calling this.
+// Returns empty string if the attached script could not be fetched.
+String ResourceLoaderText::get_attached_script_path(Ref<FileAccess> p_f) {
```

Comment:
Fun fact: this method will return the icon of the first found Script-type sub-resource, not necessarily from the `script` property. However it's nearly impossible to get it to display wrong icon without editing the scene manually, so I wouldn't worry about it.

Reply:
Yea the intention was to exit early so we avoid parsing the whole resource file. I couldn't find any use case of having multiple Scripts, so that's why it's like that right now.

Reply:
But is there a way to check if a sub-resource maps to the `script` property?

Reply:
Ah, it looks like the script property just located on the bottom of the `[resource]`. I'll have to change the parsing to account for that.

Reply:
As I said, this is not important in most cases. When you create a custom resource the script will always be first

Reply:
I think it doesn't happen very often, but I don't use custom resources much, so maybe I'm wrong.

If you find a way to fix this without affecting performance and complicating the code, go ahead. If not, I'd just leave it and wait until someone opens an issue about it. There's a chance no one will run into this situation or if they do, they won't consider it a bug (it doesn't really affect anything).

Reply:
Could we use ClassDB::APIType to detect if a class is built-in?

Reply:
@KoBeWi

> As I said, this is not important in most cases. When you create a custom resource the script will always be first

Did you perhaps mean that the script will always be *last*? Because I have a use case in my sample project for #48201 where an exported resource property itself uses a global script class that has another global script class property as its exported property, and in *that* use case, there are 2 `[ext_resource type="Script"...]` tags with the bottom one being the `script` property's `Script` (i.e. the "attached" script). Which then results in @Atlinx's code triggering a bug in my code.

To be specific, my demo has a `Tile2D` node that has a `terrain` property of type `Terrain`. It is possible to load a `WildTerrain` resource onto it (or an `UrbanTerrain`), and the `WildTerrain` has an exported array of `NaturalResource` resources. And in this use case, if I save e.g. a `wild_terrain_1.tres` file, that file needs both `NaturalResource` and `WildTerrain`, but it grabs the wrong one b/c `WildTerrain` is at the bottom and it stops after it finds the first one.

Either way, this PR needs to be amended to fix that.